### PR TITLE
test: Migrated bluebird versioned tests to `node:test`

### LINKED ISSUE #2625: Migrate bluebird versioned tests to `node:test`

### PR DESCRIPTION
**Pull request by bizob2828**
## Description

Bluebird tests were very confusing. I tried to clean this up and separate helper and fixtures and update the `methods.tap.js` to use that. I also had to update some shared tests also used by `when`.

## How to Test

```sh
npm run versioned:internal bluebird
npm run versioned:internal when
```

## Related Issues
Closes #2625

### CODE REVIEW

File: test/lib/agent_helper.js [hunk 1]
```diff
@@ -250,9 +250,15 @@ helper.unloadAgent = (agent, shimmer = require('../../lib/shimmer')) => {
 
 helper.loadTestAgent = (t, conf, setState = true) => {
```
Comment (jsumners-nr) [inline test/lib/agent_helper.js:251]: We have so many of these methods that do almost the same thing. It's very confusing to know which one to use.

File: test/lib/promises/fixtures.js
Comment (jsumners-nr) [inline test/lib/promises/fixtures.js:1]: This file is named `fixtures.js`. It doesn't look like "fixtures" to me. Maybe, "common-tests.js"? I'm not sure. It's difficult to follow all of the paths.

File: test/versioned/bluebird/fixtures.js
Comment (jsumners-nr) [inline test/versioned/bluebird/fixtures.js:1]: Same sort of comments as the other "fixtures" file.

File: test/versioned/bluebird/helpers.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+ */
+
+'use strict'
+const assert = require('node:assert')
+const { runMultiple } = require('../../lib/promises/helpers')
+const { tspl } = require('@matteo.collina/tspl')
+const symbols = require('../../../lib/symbols')
+const helper = require('../../lib/agent_helper')
+const util = require('util')
+const setImmediatePromisified = util.promisify(setImmediate)
```
Comment (jsumners-nr) [inline test/versioned/bluebird/helpers.js:13]:
```js
const { setImmediate } = require('timers/promises')
```

File: test/versioned/bluebird/helpers.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      })
+      .then(
+        function () {
+          plan.ok(!agent.getTransaction(), name + 'has no transaction')
+          testInTransaction()
+        },
+        function (err) {
+          if (err) {
+            /* eslint-disable no-console */
+            console.log(err.stack)
```
Comment (jsumners-nr) [inline test/versioned/bluebird/helpers.js:86]: Will this, and the other `console.log`s in this file, literally write to stdout during test runs? Or will they write to a trapped stream?

**Review by jsumners-nr — COMMENTED** [review]
A few questions, but nothing really blocking.

File: test/lib/agent_helper.js (hunk 1 above)
Comment (bizob2828) [reply test/lib/agent_helper.js:251]: agreed. I didn't know this one even existed until these tests.

File: test/lib/promises/fixtures.js
Comment (bizob2828) [reply test/lib/promises/fixtures.js:1]: yea I can update. I realized after that they weren't fixtures.

File: test/versioned/bluebird/fixtures.js
Comment (bizob2828) [reply test/versioned/bluebird/fixtures.js:1]: yea I can update. I realized after that they weren't fixtures.

File: test/versioned/bluebird/helpers.js (hunk 3 above)
Comment (bizob2828) [reply test/versioned/bluebird/helpers.js:86]: hmm, not sure. let me dig, this is just code that was ported

File: test/lib/agent_helper.js (hunk 1 above)
Comment (bizob2828) [reply test/lib/agent_helper.js:251]: this was is really only used when you a root test that needs an agent

**Review by jsumners-nr — APPROVED** [review]

**Pull request merged**